nilearn/connectome/connectivity_matrices.py

```diff
@@ -216,21 +216,100 @@ def sym_to_vec(symmetric, discard_diagonal=False):
     -------
     output : numpy.ndarray
         The output flattened lower triangular part of symmetric. Shape is
-        (..., n_features * (n_features + 1) / 2) if discard_diagonal is False and
-        (..., (n_features - 1) * n_features / 2) otherwise.
+        (..., n_features * (n_features + 1) / 2) if discard_diagonal is False
+        and (..., (n_features - 1) * n_features / 2) otherwise.
 
 
     """
     if discard_diagonal:
         # No scaling, we directly return the values
-        tril_mask = np.tril(np.ones(symmetric.shape[-2:]), k=-1).astype(np.bool)
+        tril_mask = np.tril(np.ones(symmetric.shape[-2:]), k=-1).astype(
+            np.bool)
         return symmetric[..., tril_mask]
     scaling = np.ones(symmetric.shape[-2:])
     np.fill_diagonal(scaling, sqrt(2.))
     tril_mask = np.tril(np.ones(symmetric.shape[-2:])).astype(np.bool)
     return symmetric[..., tril_mask] / scaling[tril_mask]
 
 
+def vec_to_sym_matrix(vec, diagonal=None):
+    """Return the symmetric matrix given its flattened lower triangular part.
+    Acts on the last dimension of the array if not 1-dimensional.
+    Diagonal can be encompassed in vec or given separately. In both cases, note
+    that diagonal elements are multiplied by sqrt(2).
+
+    .. versionadded:: 0.2.6.1
+
+    Parameters
+    ----------
+    vec : numpy.ndarray, shape (..., n_columns * (n_columns + 1) /2) or
+        (..., (n_columns - 1) * n_columns / 2) if diagonal is given seperately.
+        The input array.
```

Comment:
My opinion is that most people will want to be able to feed this function lower triangular parts of matrices and one value (0) for the diagonal. Don't you think?